echopype/tests/consolidate/test_consolidate_integration.py

```diff
@@ -147,36 +151,523 @@ def _build_ds_Sv(channel, range_sample, ping_time, sample_interval):
+                    depth_value,
+                    ds_Sv["echo_range"].isel(
+                        channel=channel_index, ping_time=ping_time_index, range_sample=range_sample_index
+                    ).data + transducer_depth_value,
+                    rtol=1e-10,
+                    atol=1e-10
+                )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("file, sonar_model, compute_Sv_kwargs", [
+    (
+        "echopype/test_data/ek60/NBP_B050N-D20180118-T090228.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek60/ncei-wcsd/Summer2017-D20170620-T021537.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH1707/Reduced_D20170826-T205615.raw",
+        "EK80",
+        {"waveform_mode":"BB", "encode_mode":"complex"}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH2106/EK80/Reduced_Hake-D20210701-T131621.raw",
+        "EK80",
+        {"waveform_mode":"CW", "encode_mode":"power"}
+    )
+])
+def test_add_depth_EK_with_platform_angles(file, sonar_model, compute_Sv_kwargs):
+    """Test `depth` values when using EK Platform angles to compute it."""
+    # Open EK Raw file and Compute Sv
+    ed = ep.open_raw(file, sonar_model=sonar_model)
+    ds_Sv = ep.calibrate.compute_Sv(ed, **compute_Sv_kwargs)
+
+    # Replace any Beam Angle NaN values with 0
+    ed["Platform"]["pitch"] = ed["Platform"]["pitch"].fillna(0)
+    ed["Platform"]["roll"] = ed["Platform"]["roll"].fillna(0)
+
+    # Compute `depth` using platform angle values
+    ds_Sv = ep.consolidate.add_depth(ds_Sv, ed, use_platform_angles=True)
+
+    # Check history attribute
+    history_attribute = ds_Sv["depth"].attrs["history"]
+    history_attribute_without_time = history_attribute[33:]
+    assert history_attribute_without_time == (
+        ". depth` calculated using: Sv `echo_range`, Echodata `Platform` Angles."
+    )
+
+    # Compute transducer depth
+    echo_range_scaling = ek_use_platform_angles(ed["Platform"], ds_Sv["ping_time"])
+
+    # Iterate through every ping time
+    for ping_time_index in range(len(ds_Sv["ping_time"])):
+        # Grab echo range scaling value
+        echo_range_scaling_value = echo_range_scaling.isel(ping_time=ping_time_index)
+        # Iterate through every channel
+        for channel_index in range(len(ds_Sv["channel"])):
+            # Grab vectors
+            depth_vector = ds_Sv["depth"].isel(
+                channel=channel_index, ping_time=ping_time_index
+            )
+            echo_range_vector = ds_Sv["echo_range"].isel(
+                channel=channel_index, ping_time=ping_time_index
+            )
+            # Check if depth vector is equal to echo range scaling value * echo range
+            assert np.all(
+                np.isclose(
+                    (depth_vector).data,
+                    (echo_range_scaling_value * echo_range_vector).data, 
+                    equal_nan=True
+                )
+            )
```

Comment:
Similar to [my comment above](https://github.com/OSOceanAcoustics/echopype/pull/1319/files#r1666435907), but it is also possible that i missed something.

Reply:
also applies to the same block under `test_add_depth_EK_with_beam_angles`